scripts/enrich_match_data.py

```diff
@@ -0,0 +1,242 @@
+        except DisambiguationError:
+            return ''
+        except PageError:
+            return ''
+        except Exception:
+            print('Unknown exception!')
+            return ''
+
+    def search_term(self, query_text: str):
+        """
+        Search query text in wikipedia
+        :param query_text:
+        :return:
+        """
+        if query_text not in self.wiki_dict:
+            try:
+                self.wiki_dict[query_text] = wikipedia.search(query_text, suggestion=True)
+            except Exception:
+                self.wiki_dict[query_text] = ([], None)
+        return self.wiki_dict[query_text]
+
+    def add_definition_to_entity(self, ent_name: str):
+        """
+        Add synonyms to entity
+        :param ent:
+        :return:
+        """
+        wiki_ents = []
+        definition = ""
+
+        try:
+            # normalize name
+            norm_name = string_utils.remove_stop(ent_name, self.tokenizer, self.STOP)
+
+            # get definition from wikipedia
+            definition = self.get_summary(norm_name)
+
+            # get suggested articles from wikipedia
+            articles, suggestion = self.search_term(norm_name)
+            wiki_ents = articles
+
+            # get definition from suggested articles if no definition retrieved earlier
+            if len(definition) <= 5 and suggestion:
+                definition = self.get_summary(suggestion)
+        except Exception:
+            return wiki_ents, definition
+
+        return wiki_ents, definition
+
+
+# class for enriching data from KB or training data
+class DataEnricher:
+    def __init__(self, data_path: str, out_path: str):
+        """
+        Initialize class for enriching data. The data_path can point to a particular knowledgebase
+        to be enriched, or it can point to a training data file.
+        :param data_path:
+        :param out_path:
+        :param restrict_path: path to list of entity ids to enrich
+        """
+        self.data_path = data_path
+        self.out_path = out_path
+
+        # clear out file
+        if os.path.exists(self.out_path):
+            open(self.out_path, 'w').close()
+
+        # initialize enricher classes
+        self.syn_enricher = SynonymEnricher()
+        self.wiki_enricher = WikipediaEnricher()
+
+    def query_all_kb(self, kb: KnowledgeBase):
+        """
+        Iterate through KB entities, query synonyms and definition, write to file.
+        :param kb:
+        :return:
+        """
+        for ent in tqdm.tqdm(kb.entities, total=len(kb.entities)):
+            mesh_syn, dbp_syn = self.syn_enricher.add_synonyms_to_entity(ent.aliases)
+            wiki_ents, definition = self.wiki_enricher.add_definition_to_entity(ent.canonical_name)
+            ent.additional_details['mesh_synonyms'] = mesh_syn
+            ent.additional_details['dbpedia_synonyms'] = dbp_syn
+            ent.additional_details['wiki_entities'] = wiki_ents
```

Comment:
It's not clear why `wiki_ents` is useful. Do we use it anywhere? As far as I can tell, the wikipedia definition may or may not come from the pages in `wiki_ents`. I find this to be error-prone/confusing.